helper/clientInitialize.go

```diff
@@ -24,30 +24,86 @@ func getAppknoxAccessToken() string {
 	return accessToken
 }
 
+// GetHostMappings returns a map of host names to URLs.
+ func GetHostMappings() map[string]string {
+     return map[string]string{
+         "global": "https://api.appknox.com/",
+         "saudi":   "https://sa.secure.appknox.com/",
+         // Add more mappings as needed
+     }
+ }
+
+func getAPIHostMappings() map[string]string {
+    // Instead of using an environment variable, call the new function
+    return GetHostMappings()
+}
+
 func getClient() *appknox.Client {
-	token := getAppknoxAccessToken()
-	host := viper.GetString("host")
-	client, err := appknox.NewClient(token)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	proxyURL, err := GetProxy()
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	insecure := viper.GetBool("insecure")
-	client = client.SetHTTPTransportParams(proxyURL, insecure)
-	baseHost, err := url.Parse(host)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	client.BaseURL = baseHost
-	return client
+    token := getAppknoxAccessToken()
+
+    // Check for region and host first
+    region := viper.GetString("region")
+    host := viper.GetString("host")
+
+    // Get the API host mappings
+    hostMappings := getAPIHostMappings()
+
+    // If host is empty, that means it was not explicitly provided, so we check the region
+    if host == "" {
+        if region != "" {
+            // Check if region exists in the mappings
+            if mappedHost, exists := hostMappings[region]; exists {
+                host = mappedHost
+            } else {
+                // Invalid region, throw error and show available regions
+                availableRegions := make([]string, 0, len(hostMappings))
+                for key := range hostMappings {
+                    availableRegions = append(availableRegions, key)
+                }
+                fmt.Printf("Invalid region name: %s. Available regions: %s\n", region, strings.Join(availableRegions, ", "))
+                os.Exit(1)
+            }
+        } else {
+            // If neither host nor region are provided, default to the global host
+            host = hostMappings["global"]
+        }
+    } else {
+        // If both region and host are provided, prioritize host and ignore region
```

Comment:
can we move this logic to a separate function along with tests? There is too much nested if here without tests?

the function should probably accept the `host` env and `region` env and return (baseurl, error) with similar logic.

Reply:
Yes, moved the logic to a separate function.